tools/eslint-rules/restrict-let-in-for-loops.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * @fileoverview Prohibit the use of `let` in `for` loops.
```

Comment:
@Trott Definitely agree with you. I changed the wording to have `Prohibit the use of `let` as the loop variable in the initialization of for, and the left-hand iterator in forIn and forOf loops.` Which may be too verbose? I just missed @TheAlphaNerd comment before changing this. Let me know what you think!